test/test_matplotlib.py

```diff
@@ -28,3 +32,51 @@ def test_pdf(selenium):
     selenium.run("import io")
     selenium.run("fd = io.BytesIO()")
     selenium.run("plt.savefig(fd, format='pdf')")
+
+
+def test_rendering(selenium):
+    selenium.load_package("matplotlib")
+    selenium.run("""
+    from js import XMLHttpRequest, window
+    window.testing = True
+    from matplotlib import pyplot as plt
+    from matplotlib import _png
+    import numpy as np
+    import io
+    t = np.arange(0.0, 2.0, 0.01)
+    s = 1 + np.sin(2 * np.pi * t)
+    plt.plot(t, s, linewidth=1.0, marker=11)
+    plt.plot(t, t)
+    plt.grid(True)
+    plt.show()
+    canvas_data = plt.gcf().canvas.get_pixel_data()
+    """)
+    selenium.get_canvas_data()
+    selenium.run("""
+    def _get_url_async(url, cb):
+        req = XMLHttpRequest.new()
+        req.open('GET', url, True)
+        req.responseType = 'arraybuffer'
+
+        def callback(e):
+            if req.readyState == 4:
+                ref_data = cb(io.BytesIO(req.response))
+                window.result = np.mean(np.abs(canvas_data - ref_data)) < 0.1
+
+        req.onreadystatechange = callback
+        req.send(None)
+    """)
```

Comment:
Lets move this to a helper function in `backend_html5_canvas`, and maybe call it `compare_reference_image`.  There isn't any need to pass in `cb` here -- just call `_png.read_png_int` directly.